File: Iseqs.cs
using System;

namespace p3
{
    class Iseqs
    {
        //Data members
        protected uint Seed = 5;
        protected const uint ArithHop = 5;
        protected uint CurrentValue;
        protected bool ObjectOn;
        protected const int RandCeiling = 9;
        protected const int RandFloor = 0;
        protected uint P;
        protected uint NumQueries;

        public Iseqs(uint PValue)
        {
            ObjectOn = true;
            CurrentValue = Seed;
            P = PValue % 10;
            NumQueries = 0;

        }
        public Iseqs()
        {
            ObjectOn = true;
            CurrentValue = Seed;
            P = GenRandomDig();
            NumQueries = 0;

        }

        public virtual uint Query()
        {
            NumQueries++;
            uint next = 0;
            if (ObjectOn)
            {
                next = CurrentValue + ArithHop;
                CurrentValue = next;
            }
            UpdateStatus();
            return next;
        }
        protected uint GenRandomDig()
        {
            uint RandDig = (uint)rand.Next(RandCeiling, RandFloor);
            return RandDig;
        }
        protected Random rand = new Random();

        

        protected virtual void UpdateStatus()
        {
            if (NumQueries % P == 0 && NumQueries !=0)
            {
                bool status = ObjectOn;
                if (status)
                    ObjectOn= false;
                else
                    ObjectOn =true;
            }
            return;
        }

        public bool Status()
        {
            return ObjectOn;
        }

    }

}

File: Iseqs.cpp
//Iseqs.cpp
//Sean Wright
//Version 4
//Main Isequence file
#include "Iseqs.h"

using namespace std;

Iseqs :: Iseqs(){}
Iseqs :: Iseqs(const Iseqs &a){}

//Preconditions: none
//Postconditions: Creates an Iseq object, set to active with an encapsulated digit
Iseqs :: Iseqs(int PValue)
{ 
	ObjectOn = true;
	CurrentValue = Seed;
	if (PValue < 1)
		PValue = 5;
	if(PValue %MakeDigit ==0)
		PValue = 7;
	P = PValue % MakeDigit;
	NumQueries = 0;
}

//Preconditions: none
//Postconditions: Can flip object status if the conditional is met
void Iseqs :: UpdateStatus(){
	if (NumQueries % P == 0 && NumQueries != 0)
		ObjectOn = !ObjectOn;
	return;
}

Iseqs :: ~Iseqs(){
	//May need to flush out
}

//Preconditions: none 
//Postconditions: will call updatestatus which can flip status of object
int Iseqs :: Query(int x){
	NumQueries++;
	//X is only needed for
	//RandomS sequence generation
	int next = 0;
	if (ObjectOn)
	{
		next = (CurrentValue + ArithHop);
		CurrentValue = next;
	}
	UpdateStatus();
	return next;
	
}

//Preconditions: none 
//Postconditions: none 
bool Iseqs :: Status(){
	return ObjectOn;
}